gooddata-sdk/gooddata_sdk/catalog/data_source/declarative_model/physical_model/pdm.py

```diff
@@ -0,0 +1,53 @@
+# (C) 2022 GoodData Corporation
+from __future__ import annotations
+
+from typing import Any
+
+from gooddata_metadata_client.model.declarative_model import DeclarativeModel
+from gooddata_metadata_client.model.declarative_tables import DeclarativeTables
+from gooddata_sdk.catalog.data_source.declarative_model.physical_model.table import CatalogDeclarativeTable
+
+
+class CatalogDeclarativePdm:
+    def __init__(self, pdm: CatalogDeclarativeTables = None):
+        self.pdm = pdm
+
+    @classmethod
+    def from_api(cls, entity: dict[str, Any]) -> CatalogDeclarativePdm:
+        pdm = CatalogDeclarativeTables.from_api(entity["pdm"])
+        return cls(pdm)
+
+    def to_api(self) -> DeclarativeModel:
+        kwargs: dict[str, Any] = dict()
+        if self.pdm:
+            kwargs["pdm"] = self.pdm.to_api()
+        return DeclarativeModel(**kwargs)
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, CatalogDeclarativePdm):
+            return False
+        return self.pdm == other.pdm
+
+
+class CatalogDeclarativeTables:
+    def __init__(
+        self,
+        tables: list[CatalogDeclarativeTable],
+    ):
+        self.tables = tables
+
+    @classmethod
+    def from_api(cls, entity: dict[str, Any]) -> CatalogDeclarativeTables:
+        tables = [CatalogDeclarativeTable.from_api(v) for v in entity["tables"]]
+        return cls(tables)
+
+    def to_api(self) -> DeclarativeTables:
+        kwargs: dict[str, Any] = {
+            "tables": [v.to_api() for v in self.tables],
+        }
```

Comment:
Do not place tables in kwargs it should work to place tables directly in DeclarativeTables.

Reply:
Pdm and Tables merged